service/controller/chart/v1/resource/release/update.go

```diff
@@ -2,20 +2,103 @@ package release
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/controller"
+	"k8s.io/helm/pkg/helm"
 )
 
+func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	releaseState, err := toReleaseState(updateChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if releaseState.Name != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating release %#q", releaseState.Name))
+
+		tarballURL := key.TarballURL(customObject)
+		tarballPath, err := r.helmClient.PullChartTarball(ctx, tarballURL)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		defer func() {
```

Comment:
Shouldn't it locate just below line number 28?

Reply:
That depends a bit on the implementation of `PullChartTarball`. Here it should be that only in case no error is returned the tarball should be written to the file system. So this is fine.